### ISSUE TITLE
How do we show errors?

### ISSUE THREAD

**Issue by v3rron** [meteor-useraccounts/core#505]
I'm not sure if it's a bug or just unclear explanation, but I can't seem to see any errors in error box.
I have the following configuration for AccountTemplates:
````
overrideLoginErrors: false,
continuousValidation: true,
negativeFeedback: true,
negativeValidation: true,
positiveValidation: true,
positiveFeedback: true,
showValidating: true,
````
1. When I try to login with empty email and password it shows "Login Forbidden" error box.
2. SignUp page errors look differently with inline validation errors under each field.
3. SignIn page doesn't have inline validation at all. Whenever the user is not found, even the error box is not showing. It just fails silently. I could only see the errors through console:
````
onSubmitHook: function(error, state){
    console.log(error); // User not found [403]
}
````
Here are the packages I'm using:
````
useraccounts:core          upgraded from 1.12.0 to 1.12.1
useraccounts:iron-routing  upgraded from 1.12.0 to 1.12.1
useraccounts:semantic-ui   upgraded from 1.12.0 to 1.12.1
````
Could you please advise how to show proper inline error messages or at least error box for login?
Thanks in advance

**Comment by sasikanth513** [comment]
check #500 
looks like same issue

**Comment by splendido** [comment]
Is this something connected with #279?
I thought it was solved...

**Comment by sasikanth513** [comment]
@splendido Thanks for the reply.
IN #279 the issue happens when given empty values and submits

Here in my case #500 errors are not showing when username/password is not correct.

Looks like the same issue(when username/password incorrect) here.

If I use like below on 'onSubmitHook'
```
Meteor.setTimeout(function(){ 
   AccountsTemplates.state.form.set("error", msgs); 
}, 100);
```
everything is working fine.

**Comment by v3rron** [comment]
Yeah, here it's the opposite... We can see "Login Forbidden" error when no values passed, but we can't see "User Not Found" error, when submitting with bad values

**Comment by sasikanth513** [comment]
@v3rron  

`AccountsTemplates.state.form.get("error");`

this is a reactive variable(reactdict), I think you can listen to this helper for errors

**Comment by v3rron** [comment]
@sasikanth513 thanks

**Comment by msolters** [comment]
`AccountsTemplates.state.form.get("error");` does not respond to `Error: User not found`, though.

**Comment by msolters** [comment]
In addition, the above method does not capture registration errors such as `Exception in delivering result of invoking 'ATCreateUserServer': Error: Email already exists.`

There must be a more complete way to capture all registration and login errors.  In addition, the configuration parameter `overrideLoginErrors` could be extended to all turnkey operations such as login and registration.

**Comment by yashha** [comment]
@msolters Did you found a solution?

**Comment by msolters** [comment]
Unfortunately not.  I just stopped using this package.

**Comment by yashha** [comment]
@msolters 
Ok hm.
Could you tell me which package you are using know for signin and signup.
Would be nice! :)

**Comment by craigdrayton** [comment]
Any update on this? I've been struggling to get a custom validation on a custom field to appear for a while now.

This correctly displays the error message for the custom field:
`func: function(){ return true;}`

This (listed in the docs as how to do this) fails silently:
`func: function(value){
        if (Meteor.isClient) {
            console.log("Validating username...");
            var self = this;
            Meteor.call("userExists", value, function(err, userExists){
                if (!userExists)
                    self.setSuccess();
                else
                    self.setError(userExists);
                self.setValidating(false);
            });
            return;
        }
        // Server
        return Meteor.call("userExists", value);
    },`

**Comment by craigdrayton** [comment]
Hi all, I was able to solve this by calling the following function (passing the field as a parameter) when the asynchronous validation returns an error.

```
var validationCallback = function(field){
    var error = field.errStr;
    field.setValidating(false);
    field.setError(error);

    var errList = [];

    errList.push({
        field: field.getDisplayName(),
        err: error
    });

    AccountsTemplates.state.form.set("error", errList);
    AccountsTemplates.setDisabled(false);
    field.setValidating(false);
};
```

**Comment by carsonperrotti** [comment]
@craigdrayton I am running into the same issue. Can you post a bit more of an example of how to work around this error?

**Comment by craigdrayton** [comment]
Sure, I have the following function as AccountsTemplates.addField.func for my custom field.

Meteor method 'invitationCodeIsInvalid' returns true if there is a validation error (yes, double negative, I know, but "func" expects _true_ to invalidate the field).

validationCallback() is the function I posted earlier.

```
func: function(value){

        if (Meteor.isClient) {
            var self = this;

            Meteor.call("invitationCodeIsInvalid", value, function(codeIsInvalid){
                if (!codeIsInvalid) {
                    self.setSuccess();
                    self.setValidating(false);
                } else {
                    validationCallback(self);
                }
            });
            return;
        }

        // Server
        return Meteor.call("invitationCodeIsInvalid", value);
    }
```

**Comment by stevengragg** [comment]
> Hi all, I was able to solve this by calling the following function (passing the field as a parameter) when the asynchronous validation returns an error.
> 
> ```
> var validationCallback = function(field){
>     var error = field.errStr;
>     field.setValidating(false);
>     field.setError(error);
> 
>     var errList = [];
> 
>     errList.push({
>         field: field.getDisplayName(),
>         err: error
>     });
> 
>     AccountsTemplates.state.form.set("error", errList);
>     AccountsTemplates.setDisabled(false);
>     field.setValidating(false);
> };
> ```



> Sure, I have the following function as AccountsTemplates.addField.func for my custom field.
> 
> Meteor method 'invitationCodeIsInvalid' returns true if there is a validation error (yes, double negative, I know, but "func" expects _true_ to invalidate the field).
> 
> validationCallback() is the function I posted earlier.
> 
> ```
> func: function(value){
> 
>         if (Meteor.isClient) {
>             var self = this;
> 
>             Meteor.call("invitationCodeIsInvalid", value, function(codeIsInvalid){
>                 if (!codeIsInvalid) {
>                     self.setSuccess();
>                     self.setValidating(false);
>                 } else {
>                     validationCallback(self);
>                 }
>             });
>             return;
>         }
> 
>         // Server
>         return Meteor.call("invitationCodeIsInvalid", value);
>     }
> ```
Hi, may I know where do you put this code, is it on the lib folder as well as the validationCallback?

**Comment by stevengragg** [comment]
Because I tried to add this bot to my lib folder but setError , setValidating, setSuccess are still not recognized in the client side, I was thinking this is because of the version but I am sure this is in latest one
